airflow/serialization/serialized_dag.py

```diff
@@ -111,26 +95,26 @@ def deserialize_dag(cls, encoded_dag: dict) -> "SerializedDAG":
 
             setattr(dag, k, v)
 
-        # pylint: disable=protected-access
-        keys_to_set_none = dag._serialized_fields - encoded_dag.keys() - cls._CONSTRUCTOR_PARAMS.keys()
+        keys_to_set_none = dag.get_serialized_fields() - encoded_dag.keys() - cls._CONSTRUCTOR_PARAMS.keys()
         for k in keys_to_set_none:
             setattr(dag, k, None)
 
         setattr(dag, 'full_filepath', dag.fileloc)
         for task in dag.task_dict.values():
             task.dag = dag
-            task = cast(SerializedBaseOperator, task)
+            serializable_task: SerializedBaseOperator = cast(SerializedBaseOperator, task)
```

Comment:
Do we need the type annotation and the cast? I would have hoped that one or the other is enough?

Reply:
Right